pages/hub/lists/new.tsx

```diff
@@ -96,169 +55,162 @@ const NewListCreationPage: WithPageLayout = () => {
 
       if (res.ok) {
         const data = await res.json();
-        setListId(data.id);
-        setIsSuccess(true);
+        return data;
       }
     } catch (error) {
       console.log(error);
-    } finally {
-      setCreateLoading(false);
-      setIsOpen(true);
+      return null;
     }
   };
 
-  const handleOnListCreate = () => {
-    const payload: CreateListPayload = {
-      name: title,
-      is_public: isPublic,
-      contributors: selectedContributors.map((contributor) => contributor.user_id),
-    };
+  const handleGitHubImport = async (props: { follow: boolean }) => {
+    if (!providerToken) {
+      toast({ description: "Unable to connect to GitHub! Try refreshing your auth session", variant: "warning" });
+      return;
+    }
 
-    handleCreateList(payload);
-  };
+    const req = await fetch(`https://api.github.com/user/following?per_page=10`, {
+      headers: {
+        Authorization: `Bearer ${providerToken}`,
+        "Content-type": "application/json",
+      },
+    });
 
-  const handleCopyToClipboard = async (content: string) => {
-    const url = new URL(content).toString();
-    posthog!.capture("clicked: List page link copied");
+    if (!req.ok) {
+      toast({ description: "Unable to connect to GitHub", variant: "warning" });
+      return;
+    }
 
-    try {
-      await navigator.clipboard.writeText(url);
-      toast({ description: "Copied to clipboard", variant: "success" });
-    } catch (error) {
-      console.log(error);
+    const following: { id: number; login: string }[] = await req.json();
+
+    const response = await createList({
+      name,
+      contributors: following.map((user) => user.id),
+      is_public: isPublic,
+    });
+
+    if (response) {
+      if (props.follow) {
+        const followRequests = following.map((user) =>
+          fetch(`${process.env.NEXT_PUBLIC_API_URL}/users/${user.login}/follow`, {
+            method: "PUT",
+            headers: {
+              Authorization: `Bearer ${sessionToken}`,
+            },
+          })
+        );
+
+        Promise.allSettled(followRequests);
```

Comment:
why no `await` here?

Reply:
also, take note of the conflict

Reply:
I removed the await because its an array of up to 10 follow requests. I don't think those should block navigating to the next page after the list is created. We should add a single endpoint to do bulk follows though. @jpmcb has an issue open for it